pkgs/os-specific/cygwin/default.nix

```diff
@@ -0,0 +1,26 @@
+{
+  makeScopeWithSplicing',
+  generateSplicesForMkScope,
+  attributePathToSplice ? [ "cygwin" ],
+}:
+
+let
+  otherSplices = generateSplicesForMkScope attributePathToSplice;
+in
+makeScopeWithSplicing' {
+  inherit otherSplices;
+  f =
+    self:
+    let
+      callPackage = self.callPackage;
+    in
+    {
+      w32api = callPackage ./w32api { };
+      w32api-headers = callPackage ./w32api/headers.nix { };
+
+      newlib-cygwin = callPackage ./newlib-cygwin { };
+      # this is here to avoid symlinks being made to cygwin1.dll in /nix/store
+      newlib-cygwin-nobin = callPackage ./newlib-cygwin/nobin.nix { };
```

Comment:
If `cygwin.newlib-cygwin` should never be used, I would not put it in the public API to reduce any potential confusion. So I would either let-bind it, or callPackage within the no-bin file.

Reply:
I would just put it in a `lib` output. But that makes sense if there are useful utils.

Reply:
See the very long thread in https://github.com/NixOS/nixpkgs/issues/431820

I agree while we might throw in the towel and do `$lib/bin`, we should definitely not do `$bin/bin`.